indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java

```diff
@@ -324,12 +324,9 @@ protected void reduce(
             HyperLogLogCollector.makeCollector(ByteBuffer.wrap(value.getBytes(), 0, value.getLength()))
         );
       }
-      Optional<Interval> intervalOptional = config.getGranularitySpec().bucketInterval(new DateTime(key.get()));
 
-      if (!intervalOptional.isPresent()) {
-        throw new ISE("WTF?! No bucket found for timestamp: %s", key.get());
-      }
-      Interval interval = intervalOptional.get();
+      Interval interval = config.getGranularitySpec().getSegmentGranularity().bucket(new DateTime(key.get()));
```

Comment:
It's not same because the bucket intervals can be non-consecutive in the granularity spec.
Here is a counterexample.

```java
final GranularitySpec spec = new UniformGranularitySpec(
    Granularities.DAY,
    null,
    Lists.newArrayList(
        new Interval("2012-01-08T00Z/2012-01-11T00Z"),
        new Interval("2012-01-07T00Z/2012-01-08T00Z"),
        new Interval("2012-01-03T00Z/2012-01-04T00Z"),
        new Interval("2012-01-01T00Z/2012-01-03T00Z")
    )
);

System.out.println("spec.bucketInterval(): " + spec.bucketInterval(new DateTime(2012, 1, 5, 0, 0)));
System.out.println("spec.getSegmentGranularity().bucket(): " + spec.getSegmentGranularity().bucket(new DateTime(2012, 1, 5, 0, 0)));
```

```
spec.bucketInterval(): Optional.absent()
spec.getSegmentGranularity().bucket(): 2012-01-05T00:00:00.000Z/2012-01-06T00:00:00.000Z
```